Added REMOVE PARTITIONING and ALTER TABLE t PARTITION BY (#13167)
pingcap/docs: partitioned-table.md

If you execute a Key partition management statement that is not yet supported, TiDB returns an error.

```sql
ALTER TABLE members OPTIMIZE PARTITION p2000;
```

```sql
ERROR 8200 (HY000): Unsupported optimize partition
```

## Convert a partitioned table to a non-partitioned table

> Remove partitioning from an already partitioned table:
> 
> - `ALTER TABLE members REMOVE PARTITIONING` which will alter a partitioned table to a non-partitioned table by copying all rows and recreate the indexes on-line.

Reviewer: ````suggestion
To convert a partitioned table to a non-partitioned table, you can use the following statement, which removes the partitioning, copies all rows of the table, and recreates the indexes online for the table:

```sql
ALTER TABLE <table_name> REMOVE PARTITIONING
```

For example, to convert the `members` partitioned table to the non-partitioned table, you can execute the following statement:

```sql
ALTER TABLE members REMOVE PARTITIONING
```
````

## Partition an existing table

> Partition an existing table (regardless if it is already partitioned or not):
> 
> - `ALTER TABLE t PARTITION BY ...` which will alter the table to a partitioned table by copying all rows and recreate the indexes on-line according to the partitioning scheme.

Reviewer: ````suggestion
To partition an existing non-partitioned table or modify the partition type of an existing partitioned table, you can use the following statement, which copies all rows and recreates the indexes online according to the new partition definitions:

```sql
ALTER TABLE <table_name> PARTITION BY <new partition definitions>
```
````

> Example:

Reviewer: ```suggestion
Examples:

To convert the existing `members` table to a HASH partitioned table with 10 partitions, you can execute the following statement:
```

```sql
ALTER TABLE members PARTITION BY HASH(id) PARTITIONS 10;
```

> or

Reviewer: ```suggestion
To convert the existing `member_level` table to a RANGE partitioned table, you can execute the following statement:
```

ERROR 8200 (HY000): Unsupported optimize partition
```

> 

Reviewer: @mjonss please also add a section in this PR to describe how to modify one partition type to another. Thanks.

Author: I rephrased 'Convert a non-partitioned table to a partitioned table' into 'Partition an existing table' with some clarifying text.

> ## How to add or remove partitioning from an existing table

Reviewer: ```suggestion
## Convert a partitioned table to a non-partitioned table
```

Remove partitioning from an already partitioned table:

- `ALTER TABLE t REMOVE PARTITIONING` which will alter a partitioned table to a non-partitioned table by copying all rows and recreate the indexes on-line.

> 

Reviewer: ```suggestion
## Convert a non-partitioned table to a partitioned table
```

Query OK, 0 rows affected (0.03 sec)
```

> ## Convert a partitioned table to a non-partitioned table

Reviewer: ```suggestion
### Convert a partitioned table to a non-partitioned table
```

To convert a partitioned table to a non-partitioned table, you can use the following statement, which removes the partitioning, copies all rows of the table, and recreates the indexes online for the table:

```sql
ALTER TABLE <table_name> REMOVE PARTITIONING
```

For example, to convert the `members` partitioned table to the non-partitioned table, you can execute the following statement:

```sql
ALTER TABLE members REMOVE PARTITIONING
```

> ## Partition an existing table

Reviewer: ```suggestion
## Partition an existing table or modify partition type
```

> ## Partition an existing table

Reviewer: ```suggestion
### Partition an existing table
```
